### ISSUE TITLE
icc_profile is a tuple when a buffer was expected on TIFF

### ISSUE THREAD

**Issue by greut** [python-pillow/Pillow#1509]
### What I did

```
from six import BytesIO
from PIL import Image

# https://ntnu.tind.io/record/109/files/UBT-TO-000008.tif
img = Image.open("UBT-TO-00008.tif")
buffer = BytesIO()
img.save(buffer, "png", quality=90)
```

### What happened

I'm using: Pillow (3.0.0)

```                                                                 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/invenio/.virtualenvs/maint-2.0/local/lib/python2.7/site-packages/PIL/Image.py", line 1665, in save
    save_handler(self, fp, filename)
  File "/home/invenio/.virtualenvs/maint-2.0/local/lib/python2.7/site-packages/PIL/PngImagePlugin.py", line 757, in _save
    data = name + b"\0\0" + zlib.compress(im.info["icc_profile"])
TypeError: must be string or read-only buffer, not tuple
```

### What was expected

If I'm using: Pillow (2.9.0), it works.

**Comment by radarhere** [comment]
From my testing, this seems to be a duplicate of #1462. If you'd like a fix, the latest 3.0.x branch will serve you until 3.0.1 is released, or you can pull only the changes necessary by looking at #1465.

**Comment by greut** [comment]
@radarhere Cool! Thanks mate. I should have checked more thoroughly in the closed issues.
